### PR TITLE
`Development`: Add e2e tests for git exercise participation using ssh and token

### PR DESCRIPTION
**Pull request by muradium**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context
We want to test the scenario where students make programming exercise submissions by Git using SSH.

### Description
Adds 2 E2E tests in Playwright about using RSA and ED25519 encrypted SSH keys for exercise submissions. For simplicity of test setup, SSH keys intented for using in Playwright have been generated inside `src/test/playwright/ssh-keys` folder. These keys are automatically copied to the ssh keys storage of the machine by Playwright global script `global-script.ts`, which is executed before tests are ran.

### Steps for Testing
- **Code Review**: Ensure that tests pass for valid reasons, functionality is adequately tested and check the code quality.
- **Run the tests (optional)**: Tests are executed during automatic run in CI environment. You can run them locally and check if they pass.

**Steps for running the tests:**
1. Navigate to **src/test/playwright**
2. Configure Playwright using playwright.env file based on your local setup. Current configuration should work for default Artemis setup.
3. Run `npm install && npm run playwright:setup`
4. Run the test group using one of the methods:
- Run `npx playwright test e2e/exercise/programming/ProgrammingExerciseParticipation.spec.ts -g "Programming exercise participation using secure git"`
- Run `npm run playwright:open` to open the Playwright UI, search for the "Programming exercise participation using secure git" test suite and run it

5. Confirm that the "Makes a git submission using SSH with {RSA/ED25519} key" tests pass

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

## Summary by CodeRabbit

- **New Features**
  - Introduced new configuration variables for SSH settings in the local environment.
  - Added `data-testid` attributes to various components for improved test accessibility.
  - Enhanced functionality for cloning repositories with SSH authentication.
  - Introduced methods for managing Git clone operations and copying clone URLs.
  - Added a new class for managing SSH public keys through API requests.

- **Bug Fixes**
  - Improved error handling and visibility checks for SSH key requirements in tests.

- **Documentation**
  - Updated comments and headers in configuration files for clarity.

- **Chores**
  - Added new SSH key files for testing purposes.
  - Updated `.gitignore` to include additional ignored files and organized existing entries.

### CODE REVIEW

**Review by BBesrour — DISMISSED** [review]
Tests passed, code lgtm

**Review by coolchock — APPROVED** [review]
tests passes, code looks good to me

**Review by SimonEntholzer — APPROVED** [review]
Works on CI, not sure yet if it's an issue with my setup, but locally it doesn't work for me fsr

**Review by SimonEntholzer — CHANGES_REQUESTED** [review]
Doesn't work on linux it seems

File: src/test/playwright/init/global-setup.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                fs.copyFileSync(sourceKeyPub, destKeyPub);
+                fs.chmodSync(destKeyPub, 0o644);
+                console.log(`Public SSH key ${keyName} copied.`);
+            } else {
+                console.log(`Public SSH key ${keyName} already exists, skipping.`);
+            }
+        }
+    } catch (error) {
+        console.error('Error during SSH key setup:', error);
+    }
```
Comment (SimonEntholzer) [inline src/test/playwright/init/global-setup.ts:48]: Tbh. I would prefer if scripts wouldn't interact with files and data outside the Artemis directory

**Review by SimonEntholzer — DISMISSED** [review]
Ok we've figured out why it didn't work:
- the keys should need to be added to the ssh-agent
- the `ssh-host-key-path` variable must either be *not* specified, or in the folder must be keys for the according SSH key type used by playwright

**Review by SimonEntholzer — APPROVED** [review]

**Review by BBesrour — APPROVED** [review]

**Pull request merged**